### PR TITLE
Move setup config to boot method

### PR DESCRIPTION
**Pull request by vinkla**
Also added publish method to move the config file to /config folder in a Laravel project.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
:+1: It's important this is done AFTER the user has a chance to override it.

**Comment by jasonlewis** [comment]
The publishing is done in the Laravel service provider as it will error it for Lumen installs. So you can remove that. `vendor:publish` should already work.

**Comment by GrahamCampbell** [comment]
This isn;t just for publishing.

File: src/Provider/ApiServiceProvider.php [hunk 1]
```diff
@@ -42,7 +51,11 @@ public function register()
      */
     protected function setupConfig()
     {
-        $this->mergeConfigFrom(__DIR__.'/../../config/api.php', 'api');
+        $source = realpath(__DIR__.'/../../config/api.php');
+
+        $this->publishes([$source => config_path('api.php')]);
```
Comment (jasonlewis) [inline src/Provider/ApiServiceProvider.php]: This looks an awful lot like publishing to me @GrahamCampbell.

It's already done here: https://github.com/dingo/api/blob/develop/src/Provider/LaravelServiceProvider.php#L19

**Comment by jasonlewis** [comment]
The `config_path` function isn't available in Lumen. That's why publishing isn't in this provider. I'm fine with the other changes.

**Comment by jasonlewis** [comment]
I'll fix it up on my end. Cheers!

**Pull request merged**